src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/HttpGoogleCloudStorageDAO.scala

```diff
@@ -53,9 +51,59 @@ class HttpGoogleCloudStorageDAO(
 
   val system = akka.actor.ActorSystem("system")
 
-  override def createBucket(userInfo: UserInfo, projectId: String, workspaceId: String): Unit = {
-    val inserter = getStorage(getBucketCredential(userInfo)).buckets.insert(projectId, new Bucket().setName(getBucketName(workspaceId)))
-    retry(inserter.execute,when500)
+  override def createBucket(userInfo: UserInfo, projectId: String, workspaceId: String, workspaceName: WorkspaceName): Unit = {
+    val bucketName = getBucketName(workspaceId)
+    val groups = getGroupDirectory.groups
+    insertGroup(groups,newGroup(bucketName,workspaceName,WorkspaceAccessLevel.Read)) {
```

Comment:
That would do a lot of superfluous work that then needs to be undone.  (E.g., we fail to create the readers group, but then go on to create the writers and owners groups, which we'll then just have to try to undo.)  This tail recursion bails when appropriate and avoids extra rework.  I'd like you to reconsider.